tests/unit_tests/simtel/test_simtel_config_writer.py

```diff
@@ -132,3 +134,41 @@ def test_convert_model_parameters_to_simtel_format(
     with open(Path(model_path) / value) as f:
         content = f.read()
         assert "Trigger 2 of 1" in content
+
+
+def test_get_simtel_metadata_with_model_parameters(simtel_config_writer):
+    model_parameters = {"test_param": {"value": 42, "meta_parameter": True}}
+
+    def mock_get_name(key, simulation_software, set_meta_parameter):
+        if simulation_software == "sim_telarray":
+            if set_meta_parameter:
+                return "test_param_meta"
+            return "test_param"
+        return None
+
+    with mock.patch(
+        "simtools.utils.names.get_simulation_software_name_from_parameter_name",
+        side_effect=mock_get_name,
+    ):
+        tel_meta = simtel_config_writer._get_simtel_metadata("telescope", model_parameters)
+        assert "metaparam telescope add test_param" in tel_meta
+        assert "metaparam telescope set test_param_meta=42" in tel_meta
```

Comment:
As mentioned, this and the code made me worry that we always write `add` and `set` in the cfg files, but it doesn't look like it. So, no need to worry I think.